Thorfile

```diff
@@ -0,0 +1,33 @@
+require "thor/group"
+
+module Middleman
+  class Generator < ::Thor::Group
+    include ::Thor::Actions
+
+    source_root File.expand_path(File.dirname(__FILE__))
+
+    def copy_base_template
+      directory "template", ".", exclude_pattern: /\.DS_Store$/
+    end
+
+    def provide_setup_script
+      run "chmod a+x bin/setup"
+    end
+
+    def run_setup_script
+      puts "Running the setup script"
+      run "bin/setup"
+    end
+
+    def initialize_git_repository
+      puts "Creating Git repository"
+      run "git init"
```

Comment:
Should we also run `git add .` and `git commit -m "Initial commit"`?

Reply:
If you're looking to follow the suspenders model, it does not run `git add` or `git commit` on the results. This makes sense to me because if you'd like to do any other things as part of an "initial commit", you can without having to amend history.

Reply:
Great point! I’ve been going back-and-forth on this, so thanks for that insight. I think I’ll make a note in the readme to indicate that a repo is initialized but the next steps like `add`ing and `commit`ing are all yours.